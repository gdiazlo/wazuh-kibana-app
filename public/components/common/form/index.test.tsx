import React from 'react';
import { render } from '@testing-library/react';
import { InputForm } from './index';

describe('[component] InputForm', () => {
		it.each`
			inputType | value | options | rest
			${'editor'} | ${'{}'} | ${{editor: {language: 'json'}}} | ${{}}
			${'number'} | ${4} | ${undefined} | ${{}}
			${'select'} | ${'value1'} | ${{select: {choices: [{label: 'Label1', value: 'value1'}, {label: 'Label2', value: 'value2'}]}}} | ${{}}
			${'switch'} | ${true} | ${{switch: {values: {enabled: {label: 'Enabled', value: true}, disabled: {label: 'Disabled', value: false}}}}} | ${{}}
			${'text'} | ${'test'} | ${undefined} | ${{isInvalid: false}}
		`('Renders correctly to match the snapshot: Input: $inputType', ({ inputType, value, options }) => {
			const wrapper = render(
				<InputForm
					type={inputType}
					value={value}
					onChange={() => {}}
					options={options}
				/>
			);
			expect(wrapper.container).toMatchSnapshot();
		});
});

