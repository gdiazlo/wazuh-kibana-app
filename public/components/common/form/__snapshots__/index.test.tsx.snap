// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component] InputForm Renders correctly to match the snapshot: Input: editor 1`] = `
<div>
  <div
    class="euiCodeEditorWrapper"
    data-test-subj="codeEditorContainer"
    style="width: 100%; height: 50px;"
  >
    <div
      class="euiCodeEditorKeyboardHint"
      data-test-subj="codeEditorHint"
      id="ia60ba5b2-38e9-11ed-9933-f589929a8b66_codeEditor"
      role="button"
      tabindex="0"
    >
      <p
        class="euiText"
      >
        Press Enter to start editing.
      </p>
      <p
        class="euiText"
      >
        When you're done, press Escape to stop editing.
      </p>
    </div>
    <div
      class=" ace_editor ace-tm"
      id="ia60ba5b1-38e9-11ed-9933-f589929a8b66"
      style="width: 100%; height: 50px;"
    >
      <textarea
        autocapitalize="off"
        autocorrect="off"
        class="ace_text-input"
        spellcheck="false"
        style="opacity: 0;"
        tabindex="-1"
        wrap="off"
      />
      <div
        aria-hidden="true"
        class="ace_gutter"
      >
        <div
          class="ace_layer ace_gutter-layer ace_folding-enabled"
        />
        <div
          class="ace_gutter-active-line"
        />
      </div>
      <div
        class="ace_scroller"
      >
        <div
          class="ace_content"
        >
          <div
            class="ace_layer ace_print-margin-layer"
          >
            <div
              class="ace_print-margin"
              style="left: 4px; visibility: visible;"
            />
          </div>
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_text-layer"
            style="padding: 0px 4px;"
          />
          <div
            class="ace_layer ace_marker-layer"
          />
          <div
            class="ace_layer ace_cursor-layer ace_hidden-cursors"
          >
            <div
              class="ace_cursor"
            />
          </div>
        </div>
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-v"
        style="display: none; width: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="width: 20px;"
        />
      </div>
      <div
        class="ace_scrollbar ace_scrollbar-h"
        style="display: none; height: 20px;"
      >
        <div
          class="ace_scrollbar-inner"
          style="height: 20px;"
        />
      </div>
      <div
        style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: hidden;"
      >
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        />
        <div
          style="height: auto; width: auto; top: 0px; left: 0px; visibility: hidden; position: absolute; white-space: pre; overflow: visible;"
        >
          XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: number 1`] = `
<div>
  <div
    class="euiFormControlLayout euiFormControlLayout--fullWidth"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <input
        class="euiFieldNumber euiFieldNumber--fullWidth"
        type="number"
        value="4"
      />
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: select 1`] = `
<div>
  <div
    class="euiFormControlLayout"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <select
        class="euiSelect"
      />
      <div
        class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
      >
        <span
          class="euiFormControlLayoutCustomIcon"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiFormControlLayoutCustomIcon__icon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: switch 1`] = `
<div>
  <div
    class="euiSwitch"
  >
    <button
      aria-checked="true"
      aria-labelledby="ia616c943-38e9-11ed-9933-f589929a8b66"
      class="euiSwitch__button"
      id="ia616c941-38e9-11ed-9933-f589929a8b66"
      role="switch"
      type="button"
    >
      <span
        class="euiSwitch__body"
      >
        <span
          class="euiSwitch__thumb"
        />
        <span
          class="euiSwitch__track"
        >
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiSwitch__icon"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
          <svg
            aria-hidden="true"
            class="euiIcon euiIcon--medium euiIcon-isLoading euiSwitch__icon euiSwitch__icon--checked"
            focusable="false"
            height="16"
            role="img"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          />
        </span>
      </span>
    </button>
    <span
      class="euiSwitch__label"
      id="ia616c943-38e9-11ed-9933-f589929a8b66"
    >
      Enabled
    </span>
  </div>
</div>
`;

exports[`[component] InputForm Renders correctly to match the snapshot: Input: text 1`] = `
<div>
  <div
    class="euiFormControlLayout euiFormControlLayout--fullWidth"
  >
    <div
      class="euiFormControlLayout__childrenWrapper"
    >
      <input
        class="euiFieldText euiFieldText--fullWidth"
        type="text"
        value="test"
      />
    </div>
  </div>
</div>
`;
